### PR TITLE
Angry Tock

### PR DESCRIPTION
**Pull request by mgwalker**

### CODE REVIEW

**Comment by mgwalker** [comment]
Production Tock API does not yet support this new functionality, but it's safe to merge because the AngryTock bot won't activate until the Tock token is configured in cloud.gov.

File: scripts/angry-tock.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+ */
+const getSlackUsers = async robot =>
+  new Promise((resolve, reject) => {
+    robot.adapter.client.web.users.list((err, response) => {
+      if (err) {
+        return reject(new Error(err));
+      }
+      return resolve(response.members);
+    });
+  });
```
Comment (adunkman) [inline scripts/angry-tock.js:62]: You can leverage [`util.promisify`](https://nodejs.org/dist/latest/docs/api/util.html#util_util_promisify_original) if you’re interested — 

```javascript
const { promisify } = require('util');

const getSlackUsers = async robot => {
  const response = await promisify(robot.adapter.client.web.users.list)();
  return response.members;
};
```

… or perhaps too far: 

```javascript
const { promisify } = require('util');

const getSlackUsers = async robot => 
  (await promisify(robot.adapter.client.web.users.list)()).members;
```

Comment (adunkman) [reply scripts/angry-tock.js:62]: (the same technique could be repeated below as well, but won’t repeat myself)

File: test/scripts/angry-tock.js [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+    angryTock = loadAngryTock();
+    angryTock(robot);
+    expect(robot.logger.warning.called).to.equal(true);
+    expect(scheduler.scheduleJob.called).to.equal(false);
+  });
+
+  describe('schedules the next shouting match on startup', () => {
+    it('if it is shouty day and before first-shout time, schedules a first-shout', () => {
+      // Monday, April 8, 1974: Hank Aaron hits his 715th career homerun,
+      // breaking Babe Ruth's record.
```
Comment (adunkman) [inline test/scripts/angry-tock.js:301]: 👏

File: scripts/angry-tock.js (hunk 1 above)
Comment (mgwalker) [reply scripts/angry-tock.js:62]: TIL Node has a built-in promisify! I'm going to switch over to that for maintainability. Thanks! 😄

**Pull request merged**